src/lib/extend/safe_curl.ts

```
@@ -0,0 +1,40 @@
+export type HttpClientOptions = HttpClientParameters[1] & { checkAddress?: SSRFCheckAddressFunction };
+export type HttpClientRequestReturn = ReturnType<HttpClient['prototype']['request']>;
+
+/**
+ * safe curl with ssrf protect
+ * @param {String} url request url
+ * @param {Object} options request options
+ * @return {Promise} response
+ */
+export async function safeCurlForApplication(this: EggCore, url: HttpClientRequestURL, options: HttpClientOptions = {}): HttpClientRequestReturn {
+  // eslint-disable-next-line @typescript-eslint/no-this-alias
+  const app = this;
```

Biome (1.9.4)

[error] This aliasing of this is unnecessary.

Arrow functions inherits `this` from their enclosing scope.
Safe fix: Use this instead of an alias.


(lint/complexity/noUselessThisAlias)